### ISSUE TITLE
Problem of testing

### ISSUE THREAD

**Issue by xiaofengWang-CCNU** [MCG-NJU/TDN#55]
For 3 crops, 10 clips: the following script should be run 10 times, Is `--weights <your_checkpoint_path>` always the same? If  `--weights <your_checkpoint_path>`  does not change, the result of testing remains the same. If  `--weights <your_checkpoint_path>`   changes with each clips, Does that mean i need to train the network 10 times.

`CUDA_VISIBLE_DEVICES=0 python3 test_models_three_crops.py  kinetics \
--archs='resnet50' --weights <your_checkpoint_path>  --test_segments=8 \
--test_crops=3 --batch_size=16 --full_res --gpus 0 --output_dir <your_pkl_path>  \
-j 4 --clip_index <your_clip_index>`

**Comment by yztongzhan** [comment]
Hi @xiaofengWang-CCNU, `--weights <your_checkpoint_path>`  is indeed the same.  However, the results of testing are **different**. Because we use sampling strategy of TSN for testing. You can check the code for more details. Others got the final results successfully [here](https://github.com/MCG-NJU/TDN/issues/49#issuecomment-996415094).

**Comment by xiaofengWang-CCNU** [comment]
> Hi @xiaofengWang-CCNU, `--weights <your_checkpoint_path>` is indeed the same. However, the results of testing are **different**. Because we use sampling strategy of TSN for testing. You can check the code for more details. Others got the final results successfully [here](https://github.com/MCG-NJU/TDN/issues/49#issuecomment-996415094).

Thank you for your reply. I checked the `dataset.py` and  got it. I've trained HMDB51 with the following commend:
`python -m torch.distributed.launch --master_port 12347 --nproc_per_node=1 main.py  hmdb51 RGB --arch resnet50 --num_segments 16 --gd 20 --lr 0.02 --lr_scheduler step  --lr_steps 50 75 90 --epochs 100 --batch-size 8 --wd 1e-4 --dropout 0.5 --consensus_type=avg --eval-freq=1 -j 4 --npb --tune_from ./best.pth.tar`
the validation Results: Prec@1 64.784 Prec@5 88.341 Loss 1.88063
the testing Results:65.56(1 crop).
I wonder if this is normal

**Comment by yztongzhan** [comment]
To reproduce our results on UCF101 and HMDB51, you need to prepare the pre-trained model and add this argument `--tune_from='checkpoint/best.pth.tar` to your command. We use the following script for fine-tuning:
```
python -m torch.distributed.launch --master_port 12347 --nproc_per_node=8 \
main.py hmdb51 RGB --arch resnet50  --num_segments 16 --gd 20 --lr 0.002 \
--lr_steps 10 20 --epochs 25 --batch-size 16 --dropout 0.8 --consensus_type=avg \
--eval-freq=1 -j 4 --tune_from='checkpoint/best.pth.tar' --npb 
```
Then we perform 10-clip and 3-crop for accuracy evaluation protocol following [TEA](https://arxiv.org/abs/2004.01398) and [STM](https://arxiv.org/abs/1908.02486). 
Considering that you only use 1 GPU (because you set `--nproc_per_node=1`) for your experiment, and you set `batch-size 8` and `learning rate = 0.02`. I think this hyper parameter is incorrect and the batch size is too small to reproduce our results. We recommend that you modify your learning rate according to the [Linear Scaling Rule](https://arxiv.org/abs/1706.02677)

**Comment by xiaofengWang-CCNU** [comment]
> To reproduce our results on UCF101 and HMDB51, you need to prepare the pre-trained model and add this argument `--tune_from='checkpoint/best.pth.tar` to your command. We use the following script for fine-tuning:
> 
> ```
> python -m torch.distributed.launch --master_port 12347 --nproc_per_node=8 \
> main.py hmdb51 RGB --arch resnet50  --num_segments 16 --gd 20 --lr 0.002 \
> --lr_steps 10 20 --epochs 25 --batch-size 16 --dropout 0.8 --consensus_type=avg \
> --eval-freq=1 -j 4 --tune_from='checkpoint/best.pth.tar' --npb 
> ```
> 
> Then we perform 10-clip and 3-crop for accuracy evaluation protocol following [TEA](https://arxiv.org/abs/2004.01398) and [STM](https://arxiv.org/abs/1908.02486). Considering that you only use 1 GPU (because you set `--nproc_per_node=1`) for your experiment, and you set `batch-size 8` and `learning rate = 0.02`. I think this hyper parameter is incorrect and the batch size is too small to reproduce our results. We recommend that you modify your learning rate according to the [Linear Scaling Rule](https://arxiv.org/abs/1706.02677)

Thank you very much for your answers and suggestions. I will further try to get the results in the paper.
